blog/en/blog/2022/10/27/ten-use-cases-api-gateway.md

```diff
@@ -0,0 +1,180 @@
+
+[API-led](https://dzone.com/articles/mulesoft-api-led-connectivity-architectural-and-de) is an _architectural approach_ that puts APIs at the heart of communications between applications and the business capabilities they need to access, in order to consistently deliver seamless functionality across all digital channels.
+
+![api-led-connectivity.png](https://static.apiseven.com/2022/10/25/635783f5bb228.png)
+
+**API-led connectivity** refers to _the technique_ of using reusable and well-designed APIs to link data and applications which in turn it is based on **API-Led architecture**. It’s _a modern approach_ that looks at the best ways of reusing APIs to boost your innovation and move quickly in the market.  At the most basic level, API-led architecture should address things like:
+
+- Securing APIs from unauthorized access and significant security threats.
+- Ensuring that consuming applications can always find the right API endpoint.
+- Throttling and/or limiting the number of calls made to an API to ensure continuous availability.
+- Supporting capabilities such as API design, testing, continuous integration, life cycle management, monitoring, and operations, to name a few.
+- Error handling and preventing error propagation across the stack.
+- Real-time monitoring of APIs with rich analytics and insight.
+- An approach for implementing scalable and flexible business capabilities, for example, in support of microservices architectures.
+
+Let’s describe in subsequent sections each usage of an API Gateway to address common requirements/challenges that arise when adopting API-led architectures.
+
+---
+
+## API resource routing
+
+The first in the list is the **API resource routing** method which uses an API Gateway to route incoming calls based on unique resource identifiers (URIs). Implementing an API gateway as the single entry point to all services means that API consumers only have to be aware of one URL domain. In this way, it becomes the API gateway's responsibility to route the traffic to the corresponding service endpoints and also enforce any applied policies as it is depicted in the below diagram.
+
+![API resource routing](https://static.apiseven.com/2022/10/25/635788a124bf4.png)
+
+It reduces complexity on the API consumer side because they do not need to track and consume functionality from multiple HTTP endpoints in case there are many services in the system.  Also, no need to implement all cross-cutting concerns, such as authentication/authorization, throttling, and rate limiting separately for each service. Most API Gateways like [Apache APISIX](https://apisix.apache.org/docs/apisix/terminology/api-gateway/) has already these core features.
```

Comment:
>they do not need to track and consume functionality from multiple HTTP endpoints in case there are many services in the system

I don't know what it means

>Also, no need to implement all cross-cutting concerns, such as authentication/authorization, throttling, and rate limiting separately for each service

YES!!!!!!!! This should be in bold, blinking, red characters. That's THE selling point.

Reply:
> I don't know what it means

It means "the client applications do not need to consume functionality from multiple HTTP endpoints" I will rephrase it.